### PR TITLE
Docu: new logic for supported versions

### PR DESCRIPTION
**Pull request by klees**
Dear everyone,

the [Technical Board of the ILIAS society](https://docu.ilias.de/goto_docu_grp_5089.html) is currently looking for ways to use the limited development resources in our community more efficiently. One thing we stumbled upon is the long support period that we, as a community, currently grant for every ILIAS version. We understand that a long support period is helpful for many institutions in various situations. Regarding the difficulties in maintaining our current feature portfolio and code with the available coders, testers and concepters, we still recommend to shorten the support period.

We thus recommend to change our support as such:

* Every ILIAS version will be **fully supported** until the end of the year after the year it was released in. **Fully supported** means that every kind of issue that is reported for the release according to our bugfixing process is eligible for a fix.
* Every ILIAS version will then gain **security support** for an additional year after that. **Security support** means that we are fixing security issues only.

This will help to preserve our development resources in various ways:

* Coders, testers and administrators will need to have less ILIAS versions in their working memory and can focus their efforts on two versions of ILIAS: The one which was released recently and the one which is in active development.
* Requests for changes and improvements will concentrate on more recent versions (since old versions are decommissioned earlier), hence saving time of everyone involved in the conception of features and improvements.
* We will lower the bar for cleaning up and refactoring of features and code, since bugfixes for differing versions will need to be made for a shorter amount of time.

We expect that (some) service providers will answer to the demand for ILIAS versions that are supported longer than the community does (if there is such demand) by offering according services. We will be happy to alter community processes accordingly if required and feasible.

We ask everyone to support us with this proposal to preserve our limited development resources.

Best regards!

Richard Klees (@klees)
on the behalf of the Technical Board of the ILIAS Society

### CODE REVIEW

File: docs/development/supported-versions.md [hunk 1]
```diff
@@ -0,0 +1,47 @@
+# Supported Versions
+
+Every ILIAS version will be **fully supported** until the end of the year after
+the year it was released in. **Fully supported** means that every kind of issu
```
Comment (mjansenDatabay) [inline docs/development/supported-versions.md]:
```suggestion
the year it was released in. **Fully supported** means that every kind of issue
```

File: docs/development/supported-versions.md [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+that is reported for the release according to our bugfixing process is eligible
+for a fix. *E.g.: A usability issue, reported for ILIAS 8 (released 2022), can
+be reported in august 2023 and is eligible for a fix then.*
```
Comment (mjansenDatabay) [inline docs/development/supported-versions.md]:
```suggestion
be reported in August 2023 and is eligible for a fix then.*
```

File: docs/development/supported-versions.md [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+
+Every ILIAS version will then gain **security support** for an additional year
+after that. **Security support** means that we are fixing security issues only.
+*E.g.: A security issue, reported for ILIAS 8 (release 2022), can be reported
+august 2024 and is eligible for a fix then. A malfunction that does make the
```
Comment (mjansenDatabay) [inline docs/development/supported-versions.md]:
```suggestion
August 2024 and is eligible for a fix then. A malfunction that does make the
```

File: docs/development/supported-versions.md [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+
+Every ILIAS version will be **fully supported** until the end of the year after
+the year it was released in. **Fully supported** means that every kind of issu
+that is reported for the release according to our bugfixing process is eligible
+for a fix. *E.g.: A usability issue, reported for ILIAS 8 (released 2022), can
+be reported in august 2023 and is eligible for a fix then.*
+
+Every ILIAS version will then gain **security support** for an additional year
+after that. **Security support** means that we are fixing security issues only.
+*E.g.: A security issue, reported for ILIAS 8 (release 2022), can be reported
```
Comment (mjansenDatabay) [inline docs/development/supported-versions.md:11]:
```suggestion
*E.g.: A security issue, reported for ILIAS 8 (release 2022), can be reported in
```

**Comment by klees** [comment]
Thx @mjansenDatabay!

**Comment by matthiaskunkel** [comment]
Instead of "Every ILIAS version will be fully supported until the end of the year after the year it was released in. " I recommend to define a fixed period of time: "Every ILIAS version will be fully supported for 18 months after the date of the first stable release." This avoids varying lifetimes of versions. With the current rule, a version released in October has a much shorter lifetime than a version released in January. And 18 months is already a much shorter periold of time as the currenly usual 27 to 30 months.

**Comment by klees** [comment]
@matthiaskunkel I think we should discuss your proposal publicly today and weigh pros and cons. I do not expect that we will finally decide on this proposal today anyway. So we may well hear arguments on your point (and probably on other things as well) and then adjust this accordingly.

**Comment by oliversamoila** [comment]
Dear @klees , dear TB, 
Thank you very much for the initiative. 
As I said in the JourFixe, I think it is a good one. And yes, it needs a solution that is as easy to communicate and understand as possible. 
My ad-hoc proposal was a little differently oriented: Release ILIAS X will be maintained until 6 or 9 months after the release of ILIAS X+1. (After that would come the one-year security support.). My main aim was and still is to ensure that there is enough time for the update on the operators' side and to accompany the update in the institution.

However, after reading your proposal in its entirety, I believe that this is also the case. Furthermore, with your proposal, one does not have to constantly calculate and consider relations. 
I would like to fully support the change as described.

These changes must not result in issues reported during the support period but not yet resolved losing their validity for later releases and the need to resolve them. I still have no idea how this can be done. 

One part in the timeline still hurts me, but maybe it has to be solved somewhere else than here: The time period between release ILIAS X and the FeatureFreeze of ILIAS X+1 is one month. (In practice from ILIAS 7 to 8 it was one day. With ILIAS 9 the Feature Freese seems to be before the release of 8). I think this does not make sense and interferes with the process of improving ILIAS in its details.

Thank you !

**Comment by dsstrassner** [comment]
> Dear @klees , dear TB, Thank you very much for the initiative. As I said in the JourFixe, I think it is a good one. And yes, it needs a solution that is as easy to communicate and understand as possible. My ad-hoc proposal was a little differently oriented: Release ILIAS X will be maintained until 6 or 9 months after the release of ILIAS X+1. (After that would come the one-year security support.). My main aim was and still is to ensure that there is enough time for the update on the operators' side and to accompany the update in the institution.
> 
> However, after reading your proposal in its entirety, I believe that this is also the case. Furthermore, with your proposal, one does not have to constantly calculate and consider relations. I would like to fully support the change as described.
> 
> These changes must not result in issues reported during the support period but not yet resolved losing their validity for later releases and the need to resolve them. I still have no idea how this can be done.
> 
> One part in the timeline still hurts me, but maybe it has to be solved somewhere else than here: The time period between release ILIAS X and the FeatureFreeze of ILIAS X+1 is one month. (In practice from ILIAS 7 to 8 it was one day. With ILIAS 9 the Feature Freese seems to be before the release of 8). I think this does not make sense and interferes with the process of improving ILIAS in its details.
> 
> Thank you !

I support Olli's statements the same!

**Comment by matthiaskunkel** [comment]
Jour Fixe, 10 JAN 2022 : We highly appreciate this suggestion and accept the changed support timeline for ILIAS beginning with ILIAS version 8.

**Comment by klees** [comment]
Thanks everyone!

**Pull request merged**